rust/src/dcerpc/dcerpc.rs

```diff
@@ -980,7 +981,7 @@ impl DCERPCState {
                 self.extend_buffer(buffer, direction);
                 return AppLayerResult::ok();
             }
-            if parsed == -2 {
+            if parsed < 0 {
```

Comment:
Did you get an explanation about the QA pcap ?

Otherwise, I think we do not want to return `AppLayerResult::err();` on invalid header, but just return `AppLayerResult::ok();`, maybe set an event...

Reply:
> Did you get an explanation about the QA pcap ?
> 

Nope.

> Otherwise, I think we do not want to return `AppLayerResult::err();` on invalid header, but just return `AppLayerResult::ok();`, maybe set an event...

ok.. anomaly of invalid header?

Reply:
> > Update: Tried to set an applayer event as do other parsers for invalid data but they seems to run on transactions and this failure of header parsing is happening before a transaction is even created.
> 
> A solution may be to create a new transaction with just the event...
> 

I see.

> But to know what needs to be done, we should know what this QA pcap is...

😢